ambry-tools/src/main/java/com.github.ambry/clustermap/HelixVcrPopulateTool.java

```diff
@@ -0,0 +1,190 @@
+    clusterSetup.addCluster(destClusterName, true);
+
+    // set ALLOW_PARTICIPANT_AUTO_JOIN
+    HelixConfigScope configScope = new HelixConfigScopeBuilder(HelixConfigScope.ConfigScopeProperty.CLUSTER).
+        forCluster(destClusterName).build();
+    Map<String, String> helixClusterProperties = new HashMap<>();
+    helixClusterProperties.put(ZKHelixManager.ALLOW_PARTICIPANT_AUTO_JOIN, String.valueOf(true));
+    destAdmin.setConfig(configScope, helixClusterProperties);
+    // set PersistBestPossibleAssignment
+    ConfigAccessor configAccessor = new ConfigAccessor(destZkClient);
+    ClusterConfig clusterConfig = configAccessor.getClusterConfig(destClusterName);
+    clusterConfig.setPersistBestPossibleAssignment(true);
+    configAccessor.setClusterConfig(destClusterName, clusterConfig);
+    System.out.println("Cluster " + destClusterName + " is created successfully!");
+    return;
+  }
+
+  /**
+   * Update dest cluster information based on src cluster.
+   * Dest cluster resource will be recreated if it mismatches that in src cluster.
+   * @param srcZkString the src cluster's zk string
+   * @param srcClusterName the src cluster's name
+   * @param destZkString the dest cluster's zk string
+   * @param destClusterName the dest cluster's name
+   * @param dryRun run the update process but without actual change.
+   */
+  static void updateResourceAndPartition(String srcZkString, String srcClusterName, String destZkString,
+      String destClusterName, boolean dryRun) {
+
+    HelixAdmin srcAdmin = new ZKHelixAdmin(srcZkString);
+    Set<String> srcResources = new HashSet<>(srcAdmin.getResourcesInCluster(srcClusterName));
+    HelixAdmin destAdmin = new ZKHelixAdmin(destZkString);
+    Set<String> destResources = new HashSet<>(destAdmin.getResourcesInCluster(destClusterName));
+
+    for (String resource : srcResources) {
+      if (ignoreResourceKeyWords.stream().anyMatch(resource::contains)) {
+        System.out.println("Resource " + resource + " from src cluster is ignored");
+        continue;
+      }
+      boolean createNewResource = false;
+      if (destResources.contains(resource)) {
+        // check if every partition exist.
+        Set<String> srcPartitions = srcAdmin.getResourceIdealState(srcClusterName, resource).getPartitionSet();
+        Set<String> destPartitions = destAdmin.getResourceIdealState(destClusterName, resource).getPartitionSet();
+        if (srcPartitions.size() != destPartitions.size()) {
+          createNewResource = true;
+        } else {
+          for (String partition : srcPartitions) {
+            if (!destPartitions.contains(partition)) {
+              createNewResource = true;
+              break;
+            }
+          }
+        }
+        if (dryRun) {
+          System.out.println("DryRun: Drop Resource " + resource);
+        } else {
+          // This resource need to be recreate.
+          destAdmin.dropResource(destClusterName, resource);
```

Comment:
It looks like this should only be done if the resource exists in dest and is different from src.  You might need another boolean variable like dropChangedResource to condition this action.